presto-main/src/main/java/com/facebook/presto/operator/aggregation/MapMergeAggregation.java

```diff
@@ -0,0 +1,208 @@
+                    "map<K,V>", ImmutableList.of("map<K,V>"), false, false);
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Merges 2 maps";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager,
+                                   FunctionRegistry functionRegistry)
+    {
+        Type keyType = types.get("K");
+        Type valueType = types.get("V");
+        Signature signature = new Signature(NAME, new MapType(keyType, valueType).getTypeSignature(),
+                new MapType(keyType, valueType).getTypeSignature());
+        InternalAggregationFunction aggregation = generateAggregation(keyType, valueType);
+        return new FunctionInfo(signature, getDescription(), aggregation);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type keyType, Type valueType)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(MapMergeAggregation.class.getClassLoader());
+        Type outputType = new MapType(keyType, valueType);
+        List<Type> inputTypes = ImmutableList.of(outputType);
+        KeyValuePairStateSerializer stateSerializer = new KeyValuePairStateSerializer(keyType, valueType);
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, outputType, inputTypes),
+                createInputParameterMetadata(outputType),
+                INPUT_FUNCTION,
+                null,
+                null,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                KeyValuePairsState.class,
+                stateSerializer,
+                new KeyValuePairsStateFactory(keyType, valueType),
+                outputType,
+                false);
+
+        GenericAccumulatorFactoryBinder factory =
+                new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, outputType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type inputType)
+    {
+        return ImmutableList.of(new ParameterMetadata(STATE),
+                new ParameterMetadata(NULLABLE_BLOCK_INPUT_CHANNEL, inputType),
+                new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    //todo Unit tests.
+
+    public static void input(KeyValuePairsState toState, Block value, int position)
+    {
+        Type keyType = toState.getKeyType();
+        Type valueType = toState.getValueType();
+        SingleState fromState = new SingleState(keyType, valueType);
+        try {
+            //Todo how do we avoid creating a new instance of this?
```

Comment:
Hello Presto Engineers/@electrum/@martint, would you be able to help me with this line? Is there some way I can avoid creating multiple instances of this serializer?